src/compiler/checker.ts

```diff
@@ -10449,7 +10449,10 @@ namespace ts {
                     case SyntaxKind.FunctionExpression:
                     case SyntaxKind.FunctionDeclaration:
                     case SyntaxKind.ArrowFunction:
-                    case SyntaxKind.ObjectLiteralExpression: return false;
+                    case SyntaxKind.ClassExpression:
+                    case SyntaxKind.ClassDeclaration:
+                    case SyntaxKind.GetAccessor:
+                    case SyntaxKind.SetAccessor: return false;

```

Comment:
I think that actually *is* allowed. In https://people.mozilla.org/~jorendorff/es6-draft.html#sec-super-keyword-runtime-semantics-evaluation, it talks about GetNewTarget and GetThisEnvironment, which I believe give you the environment of the constructor for A, and not the LexicalEnvironment of class C.

So I'd say isClassLike is not correct per the ES6 rules. But as @DanielRosenwasser points out, it's pretty silly, so I think disallowing it is okay, just like we do for `yield` in a similar position.